slides/advanced-01-feature-engineering.qmd

```diff
@@ -0,0 +1,783 @@
+hotel_rec <- 
+  recipe(avg_price_per_room ~ ., data = hotel_tr) %>% 
+  step_dummy(all_nominal_predictors()) %>% 
+  step_zv(all_predictors()) %>% 
+  step_normalize(all_numeric_predictors()) %>% 
+  embed::step_umap(all_numeric_predictors(), outcome = avg_price_per_room)
+```
+
+. . . 
+
+A fancy machine learning supervised dimension reduction technique...
+
+:::notes
+Note that this uses the outcome, and it is from an extension package
+:::
+
+
+## Other possible steps `r hexes("recipes")`
+
+```{r}
+#| label: splines
+#| eval: false
+#| code-line-numbers: "6"
+hotel_rec <- 
+  recipe(avg_price_per_room ~ ., data = hotel_tr) %>% 
+  step_dummy(all_nominal_predictors()) %>% 
+  step_zv(all_predictors()) %>% 
+  step_normalize(all_numeric_predictors()) %>% 
+  step_spline_natural(year_day, deg_free = 10)
+```
+
+. . . 
+
+Nonlinear transforms like natural splines, and so on!
+
+##  {background-iframe="https://recipes.tidymodels.org/reference/index.html"}
+
+::: footer
+:::
+
+
+## Your turn {transition="slide-in"}
+
+![](images/parsnip-flagger.jpg){.absolute top="0" right="0" width="150" height="150"}
+
+*Create a `recipe()` for the hotel data to :*
+
+-   *use a Yeo-Johnson (YJ) transformation on `lead_time`*
+-   *convert factors to indicator variables*
+-   *remove zero-variance variables*
+
+```{r}
+#| label: your-turn-make-recipe
+#| echo: false
+countdown(minutes = 3, id = "make-recipe")
+```
+
+
+## Minimal recipe `r hexes("recipes")` 
+
+```{r}
+#| label: hotel-rec
+hotel_indicators <-
+  recipe(avg_price_per_room ~ ., data = hotel_tr) %>% 
+  step_YeoJohnson(lead_time) %>%
+  step_dummy(all_nominal_predictors()) %>%
+  step_zv(all_predictors())
+```
+
+## Using a workflow `r hexes("recipes", "workflows", "parsnip", "tune")` {.annotation}
+
+```{r}
+#| label: lm-model
+#| cache: true
+
+set.seed(9)
+
+hotel_lm_wflow <-
+  workflow() %>%
+  add_recipe(hotel_indicators) %>%
+  add_model(linear_reg())
+ 
+ctrl <- control_resamples(save_pred = TRUE)
+hotel_lm_res <-
+  hotel_lm_wflow %>%
+  fit_resamples(hotel_rs, control = ctrl)
+
+collect_metrics(hotel_lm_res)
+```
+
+## Your turn {transition="slide-in"}
+
+![](images/parsnip-flagger.jpg){.absolute top="0" right="0" width="150" height="150"}
+
+*Use `fit_resamples()` to fit your workflow with a recipe.*
+
+*Collect the predictions from the results.*
+
+
+```{r}
+#| label: your-turn-resample-recipe
+#| echo: false
+countdown(minutes = 5, id = "resample-recipe")
+```
+
+
+## Holdout predictions `r hexes("recipes", "workflows", "parsnip", "tune")`
+
+```{r}
+#| label: goldout-pred
+# Since we used `save_pred = TRUE`
+lm_val_pred <- collect_predictions(hotel_lm_res)
+lm_val_pred %>% slice(1:7)
+```
+
+
+## Calibration Plot `r hexes("probably")`
+
+
+```{r}
+#| label: lm-cal-plot
+#| out-width: 40%
+#| fig-width: 5
+#| fig-height: 5
+#| fig-align: "center"
+
+library(probably)
+
+cal_plot_regression(hotel_lm_res, alpha = 1 / 5)
+```
+
+
+
+## What do we do with the agent and company data? 
```

Comment:
I'll make "extra" slides using some of these slides plus the effect encoding steps and workflow sets